index.js

```diff
@@ -0,0 +1,158 @@
+ *  "options":{}}]},"billing_address":{"country_code":[{"code":"not_supported",
+ *  "message":"is not supported","options":{}}]}}}}
+ *
+ * Although Shopify has started to build helper methods for error parsing, there is still work
+ * to be done there. Also, they are different for Android and SDK.
+ *
+ * This JavaScript method gives you a complete solution for catching all checkout errors for
+ * both iOS and Android. It also transforms keys into user friendly labels, for example 'last_name'
+ * to 'Last name'.
+ *
+ * If the checkout error is a network or other non standard error, this method will return an
+ * unknown error message.
+ */
+const getCheckoutError = (errorBody, cart) => {
+  if (!errorBody) {
+    return UNKNOWN_ERROR;
+  }
+
+  const errorObject = JSON.parse(errorBody);
+
+  const checkoutErrors = errorObject.errors.checkout;
+  // This method is only used for checkout erorrs
+  if (!checkoutErrors) {
+    return UNKNOWN_ERROR;
+  }
+
+  // We first check for line item errors, which can happen only when trying to checkout the cart
+  if (checkoutErrors.line_items) {
+    return getLineItemsErrorMessage(checkoutErrors.line_items, cart);
+  }
+
+  return getErrorMessageFromCheckoutObject(checkoutErrors);
+};
+
+/**
+ * Converts line item errors to a user friendly message. If some of the items are unavailable,
+ * Shopify will send an array of these errors, in the same order as the cart items used to start
+ * the checkout. For each error, the message will describe how many units of the selected variant
+ * are remaining in the store. Those items that can be bought are represented with a null in the array.
+ * Here is an example:
+ *
+ * {"errors":{"checkout":{"line_items":[null,{"quantity":[{"message":"Not enough items available.
+ *  Only 2 left.","options":{"remaining":2},"code":"not_enough_in_stock"}]}]}}}
+ *
+ * Since this array matches the cart in order, we can display how many items for a product
+ * and variant are remaining for each cart item that is unavailable. For example, let's say we
+ * tried to buy a Small Navy sweater and 3 Grey Medium Sweaters. Suppose that only 2 Grey Sweaters
+ * are available. We'll get null for the Navy sweater and an error message for Grey ones. The resulting
+ * message will be: "Sweater - Grey Medium: Not enough items available. Only 2 left."
+ *
+ * We only handle quantity errors since these are known and documented.
+ *
+ * Available inventory is considered sensitive data and is only available through the admin API on Shopify,
+ * and not through the SDK. This is explained by the Shopify's team here:
+ * https://ecommerce.shopify.com/c/shopify-apis-and-technology/t/mobile-buy-sdk-how-can-i-obtain-the-quantity-of-a-product-279864
+ *
+ * The checkout error is the only way to tell the user how many items are available, otherwise app
+ * developers could implement a limit in the cart interface.
+ *
+ */
+const getLineItemsErrorMessage = (lineItemErrors, cart) => {
```

Comment:
I added this question to the list we prepared for Boris and the design team. If we show quantity errors next to cart items, we'll have to change the method a bit. We'll return an array with quantity errors for each item, as you said, instead of a message. Let's have Boris confirm this first and I'll make the necessary adjustments.